### ISSUE TITLE
Docker container cannot be killed

### ISSUE THREAD

**Issue by tosan88** [coreos/bugs#1204]
Hi,

From time to time we are seeing an issue with stopping/removing a docker container. It enters in a `Removal In Progress` state and cannot be stopped/removed. 

We start the docker container via fleet service unit with the following command:

`ExecStart=/bin/sh -c "/usr/bin/docker run --rm --name=%p -P -e SYS_HC_HOST_PATH=$SYS_HC_HOST_PATH -v /:/$SYS_HC_HOST_PATH:ro coco/coco-system-healthcheck:$DOCKER_APP_VERSION"`

Here are the relevant system logs:

```
Apr 01 07:03:14 ip-172-24-185-210.eu-west-1.compute.internal systemd[1]: Started docker container 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e.
Apr 01 07:03:17 ip-172-24-185-210.eu-west-1.compute.internal sh[1491]: Container running 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e /system-healthcheck
...
Apr 01 10:23:55 ip-172-24-185-210.eu-west-1.compute.internal systemd[1]: Stopped fleet daemon.
Apr 01 10:23:55 ip-172-24-185-210.eu-west-1.compute.internal systemd[1]: Closed Fleet API Socket.
...
Apr 01 10:24:11 ip-172-24-185-210.eu-west-1.compute.internal dockerd[853]: time="2016-04-01T10:24:11.259530229Z" level=info msg="Container 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e failed to exit within 10 seconds
Apr 01 10:24:11 ip-172-24-185-210.eu-west-1.compute.internal systemd[1]: Stopped docker container 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e.
Apr 01 10:24:59 ip-172-24-185-210.eu-west-1.compute.internal docker[11786]: Failed to kill container (system-healthcheck): Error response from daemon: Cannot kill container system-healthcheck: Container 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e is not running
Apr 01 10:24:59 ip-172-24-185-210.eu-west-1.compute.internal sh[12044]: docker: Error response from daemon: Conflict. The name "/system-healthcheck" is already in use by container 69c873d0b3af96347d3e7b5f96fd640a4f0c209081420945313d3e33f0f0751e. You have to remove (or rename) that container to be able to reuse that name..
```

Notice fleet daemon was stopped before the issue happened, however I don't know if it's related.

Here is the output from `docker info`:

```
core@ip-172-24-73-234 ~ $ docker info
Containers: 33
 Running: 32
 Paused: 0
 Stopped: 1
Images: 32
Server Version: 1.10.3
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: none
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.6-coreos
Operating System: CoreOS 1000.0.0 (MoreOS)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67 GiB
Name: ip-172-24-73-234.eu-west-1.compute.internal
ID: OLAY:RLSY:Y6YB:ULVX:NDON:JIKB:QMU5:7KEU:I57S:NQQX:2QXZ:2QV5
```

This is happening on CoreOS alpha version `1000.0.0`.
This is not reproducable on CoreOS stable version `899.13.0`.

Can you please help?
Thanks!

**Comment by crawford** [comment]
Do you have a simple repro case?

**Comment by Zapadlo** [comment]
Sadly no, it seems to happen sporadically, docker logs:

```
Apr 13 14:10:01 ip-172-24-185-210.eu-west-1.compute.internal dockerd[940]: time="2016-04-13T14:10:01.278915473Z" level=error msg="Handler for POST /v1.22/containers/v1-suggestor-1/kill returned error: Cannot kill container v1-suggestor-1: No such container: v1-suggestor-1"
Apr 13 14:10:01 ip-172-24-185-210.eu-west-1.compute.internal dockerd[940]: time="2016-04-13T14:10:01.323811078Z" level=error msg="Handler for DELETE /v1.22/containers/v1-suggestor-1 returned error: No such container: v1-suggestor-1"
Apr 13 14:10:01 ip-172-24-185-210.eu-west-1.compute.internal dockerd[940]: time="2016-04-13T14:10:01.484675804Z" level=error msg="Handler for POST /v1.22/containers/create returned error: Conflict. The name \"/v1-suggestor-1\" is already in use by container b8032c090569cc0a03fe52043813396eaf86ca81b10cb27f091d3139d0a2a411. You have to remove (or rename) that container to be able to reuse that name."
Apr 13 14:10:01 ip-172-24-185-210.eu-west-1.compute.internal dockerd[940]: time="2016-04-13T14:10:01.566000045Z" level=error msg="Handler for POST /v1.22/containers/v1-suggestor-1/stop returned error: No such container: v1-suggestor-1"
```

Present in `1010.1.0`

Any logs, directories I can check to provide more information? Thanks

**Comment by crawford** [comment]
```
No such container: v1-suggestor-1"
The name \"/v1-suggestor-1\" is already in use
```

Jeez, Docker looks really confused. Have you tried filing an issue upstream? This really doesn't look specific to CoreOS.

**Comment by webwurst** [comment]
Happens to me on Fedora with Docker `1.11.0`

**Comment by Zapadlo** [comment]
Found this issue https://github.com/docker/docker/issues/21198 which seems to be the the one we are seeing.

https://github.com/docker/docker/issues/21198#issuecomment-214384991 says it doesn't happen in docker version `1.11.0`, @webwurst says the opposite.

Are there any plans to move `1.11.0` to alpha so we can try it out?

Thanks.

**Comment by Zapadlo** [comment]
This is now affecting us in Production as `1010.5.0` is now stable including docker:

```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   8acee1b
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   8acee1b
 Built:
 OS/Arch:      linux/amd64
```

**Comment by crawford** [comment]
@Zapadlo I'm trying to narrow down the root cause on this but I cannot reproduce the issue myself. Can you try using the btrfs graph driver for Docker? That will let us know if this is an overlayfs issue. We have docs detailing how to [create a btrfs volume file](https://coreos.com/os/docs/latest/mounting-storage.html#creating-and-mounting-a-btrfs-volume-file).

**Comment by Zapadlo** [comment]
@crawford 

Looks like it isn't:

```
core@ip-172-24-52-118 ~ $ df -khT
Filesystem     Type      Size  Used Avail Use% Mounted on
devtmpfs       devtmpfs  7.9G     0  7.9G   0% /dev
tmpfs          tmpfs     7.9G     0  7.9G   0% /dev/shm
tmpfs          tmpfs     7.9G  2.2M  7.9G   1% /run
tmpfs          tmpfs     7.9G     0  7.9G   0% /sys/fs/cgroup
/dev/xvda9     ext4       27G  8.4G   18G  33% /
/dev/xvda3     ext4      985M  589M  345M  64% /usr
/dev/xvda1     vfat      128M   37M   92M  29% /boot
tmpfs          tmpfs     7.9G     0  7.9G   0% /media
tmpfs          tmpfs     7.9G   20K  7.9G   1% /tmp
/dev/xvda6     ext4      108M   64K   99M   1% /usr/share/oem
/dev/xvdf      ext4      197G  140M  187G   1% /vol
/dev/loop1     btrfs      24G  4.5G   19G  20% /var/lib/docker
```

```
Jun 03 08:52:34 ip-172-24-130-22.eu-west-1.compute.internal systemd[1]: Started MongoDB.
Jun 03 08:52:34 ip-172-24-130-22.eu-west-1.compute.internal sh[25105]: docker: Error response from daemon: Conflict. The name "/mongodb-3" is already in use by container 74409c2aeb545c1ca07d32acdda55115dc767a3974044c55cf6d4f0f17a00b26. You have to remove (or rename) that container to be able to reuse that name..
Jun 03 08:52:34 ip-172-24-130-22.eu-west-1.compute.internal sh[25105]: See '/usr/bin/docker run --help'.
Jun 03 08:52:34 ip-172-24-130-22.eu-west-1.compute.internal systemd[1]: mongodb@3.service: Main process exited, code=exited, status=125/n/a
Jun 03 08:52:34 ip-172-24-130-22.eu-west-1.compute.internal docker[25139]: Failed to stop container (mongodb-3): Error response from daemon: No such container: mongodb-3
```

**Comment by cdwertmann** [comment]
We see the same problem on CoreOS 1010.1.0 every day. Quickest fix is to restart docker, then restart fleet. I would like to try docker 1.11, which has been released two months ago but hasn't made it into CoreOS alpha yet (why?).

**Comment by martingartonft** [comment]
Is there anything I can do or information I can provide to help diagnose this?

**Comment by crawford** [comment]
@martingartonft this looks like a Docker issue. It would be helpful to see if this issue is still present in 1.11.2 (currently in Alpha).

**Comment by crawford** [comment]
Upstream issue: https://github.com/docker/docker/issues/23371.

**Comment by gjonespf** [comment]
Had same issue today with containers stuck in "Removal in progress".  Not sure of root cause.

```
docker ps -a | grep "Removal in progress"
954fac97bef9        redis                         "docker-entrypoint.sh"   10 days ago         Removal In Progress

```

Workaround that worked for me was:

```
sudo systemctl stop docker
sudo rm -Rf /var/lib/docker/containers/954fac97bef9--rest-of-contid--
sudo systemctl start docker
sudo systemctl restart fleet
```

Certainly seems like it's the upstream issue.

**Comment by crawford** [comment]
According to upstream, this has been fixed in Docker 1.12. Closing.
